src/server/svr_chk_owner.c

```diff
@@ -388,68 +389,94 @@ authenticate_user(struct batch_request *preq, struct connection *pcred)
- * @par
- *		Depending on what the "jobid" identifies, the following is returned
- *		in the integer pointed to by rc:
- *	   	IS_ARRAY_NO (0)       - for a regular job
- *	   	IS_ARRAY_ArrayJob (1) - for an Array Job
- *	   	IS_ARRAY_Single (2)   - for a single subjob
- *	   	IS_ARRAY_Range (3)    - for a range of  subjobs
+ *		Job Array), also returns pointer to the parent Array Job and the offset of
+ *		the subjob in its parent array.
  *
  * @param[in]	jobid	-	Job Id.
- * @param[in,out]	preq	-	job batch request
- * @param[out]	rc	-	Depending on what the "jobid" identifies,
- * 						the following is returned
- *						(0)       - for a regular job
- *						(1) - for an Array Job
- *						(2)   - for a single subjob
- *						(3)    - for a range of  subjobs
- *
- * @return	job *
- * @retval	a pointer to the job	: if found and the tests pass.
- * @retval	NULL	: failed
+ * @param[in,out]	preq	-	job batch request (cannot be NULL)
+ * @param[in,out]	jobtype	-	Depending on what the "jobid" identifies,
+ * 						the following is returned (cannot be NULL)
+ *						IS_ARRAY_NO (0)       - for a regular job
+ *						IS_ARRAY_ArrayJob (1) - for an Array Job
+ *						IS_ARRAY_Single (2)   - for a single subjob
+ *						IS_ARRAY_Range (3)    - for a range of  subjobs
+ * @param[in,out]	jobstate	-	NULL or pointer to the state of the job
+ * @param[in,out]	ppjob	-	pointer to pointer to the job (cannot be NULL)
+ * @param[in,out]	parent	-	NULL or pointer to pointer to the parent job of subjob
+ * @param[in,out]	poffset	-	NULL or offset of subjob in tracking table
+ *
+ * @return	int
+ * @retval	TRUE	: jobid is valid and the tests pass.
+ * @retval	FALSE	: jobid is invalid or tests failed
  */
 
 
-job *
-chk_job_request(char *jobid, struct batch_request *preq, int *rc)
+int
+chk_job_request(char *jobid, struct batch_request *preq, int *jobtype, int *jobstate, job **ppjob, job **parent, int *poffset)
 {
-	int	 t;
+	int offset;
 	int	 histerr = 0;
 	job	*pjob;
 	int deletehist = 0;
 	char	*p1;
 	char	*p2;
 
+	if (ppjob == NULL || jobtype == NULL ) {
+		log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_SERVER, LOG_ERR,
+			jobid, msg_system);
+		req_reject(PBSE_SYSTEM, 0, preq);
+		return FALSE;
+	}
+
 	if (preq->rq_extend && strstr(preq->rq_extend, DELETEHISTORY))
 		deletehist = 1;
-	t = is_job_array(jobid);
-	if ((t == IS_ARRAY_NO) || (t == IS_ARRAY_ArrayJob))
-		pjob = find_job(jobid);		/* regular or ArrayJob itself */
+
+	*jobtype = is_job_array(jobid);
+
+	if (*jobtype != IS_ARRAY_Range)
+		pjob = *ppjob = find_job(jobid);		/* regular or subjob or ArrayJob itself */
 	else
-		pjob = find_arrayparent(jobid); /* subjob(s) */
+		pjob = *ppjob = find_arrayparent(jobid); /* range of subjob(s) */
+
+	if (parent) {
+		if (( *jobtype == IS_ARRAY_Single ) || ( *jobtype == IS_ARRAY_Range )) {
+			if	((*parent = find_arrayparent(jobid)) == NULL) {
+				goto err_unknown_jid;
+			}
+		}else
+			*parent = NULL;
+	}
 
-	*rc = t;
 
 	if (pjob == NULL) {
```

Comment:
Update the comments to reflect why pjob can be NULL here.

Reply:
This function has been reverted in rc 20 https://github.com/PBSPro/pbspro/pull/590/commits/b01fcac3e94363966f635e6fad726a5342288a79